### PR TITLE
Add documentation about auth for the collector

### PR DESCRIPTION
**Pull request by jpkrohling**
Please, do not merge until https://github.com/open-telemetry/opentelemetry-collector/pull/3369 is merged.

﻿Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
@alolita, @gramidt, @pavankrish123, would you please review this one? This is an initial draft that can be expanded later.

File: content/en/docs/collector/configuration.md [hunk 1]
```diff
@@ -448,3 +448,94 @@ following proxy environment variables:
...
+Each authentication extension has two possible usages: as client authenticator for exporters,
+adding auth data to outgoing requests, and as server authenticator for receivers, authenticating
+incoming connections. Refer to the authentication extension for a list of its capabilities, but
+in general, an authentication extension would only implement one of those traits. For a list of
+known authenticators, use the [Registry](/registry/?s=authenticator&component=extension&language=) available in this website.
+
+To add a server authenticator to a receiver in your collector, make sure to:
+
+1. add the authenticator extension under the `extensions` section
+1. add the extension to the `services` node
```
Comment (gramidt) [inline content/en/docs/collector/configuration.md:471]: nit - Having both words "section" and "node" for describing configuration is confusing. Do you believe dot notation would help with describing the area where configuration is needed?

**Review by gramidt — APPROVED** [review]
Just a minor nit and everything else looks amazing. Great work!

LGTM!

**Review by mtwo — APPROVED** [review]

File: content/en/docs/collector/configuration.md (hunk 1 above)
Comment (jpkrohling) [reply content/en/docs/collector/configuration.md]: Done! PR updated.

**Comment by jpkrohling** [comment]
I'm marking this as ready, as the OAuth2 client authenticator extension has been approved.

**Pull request merged**